### PR TITLE
Add table oci_database_mysql_db_system. Closes #161

### LINKED ISSUE #161: Add table oci_mysql_db_system
**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.

### PR DESCRIPTION
**Pull request by bigdatasourav**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
SETUP: tests/oci_mysql_db_system []

PRETEST: tests/oci_mysql_db_system

TEST: tests/oci_mysql_db_system
Running terraform
oci_core_vcn.named_test_resource: Creating...
oci_core_vcn.named_test_resource: Creation complete after 1s [id=ocid1.vcn.oc1.ap-mumbai-1.amaaaaaa6igdexaaxifjmad3nv2cxh7m5acc4bdekilub5w423eqopew2aoq]
oci_core_subnet.named_test_resource: Creating...
oci_core_subnet.named_test_resource: Creation complete after 2s [id=ocid1.subnet.oc1.ap-mumbai-1.aaaaaaaazy7lyqmivjjsddvipxs6whkfsiczbg4j6wrt6ki6cs2q4vqkopaq]
oci_mysql_mysql_db_system.named_test_resource: Creating...
oci_mysql_mysql_db_system.named_test_resource: Still creating... [10s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [20s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [30s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [40s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [50s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [1m0s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [1m10s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [1m20s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [1m30s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [1m40s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [1m50s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [2m0s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [2m10s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [2m20s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [2m30s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [2m40s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [2m50s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [3m0s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [3m10s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [3m20s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [3m30s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [3m40s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [3m50s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [4m0s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [4m10s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [4m20s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [4m30s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [4m40s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [4m50s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [5m0s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [5m10s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [5m20s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [5m30s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [5m40s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [5m50s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [6m0s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [6m10s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [6m20s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [6m30s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [6m40s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [6m50s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [7m0s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [7m10s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [7m20s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [7m30s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [7m40s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [7m50s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [8m0s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [8m10s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [8m20s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Creation complete after 8m22s [id=ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaa3htpdhd6a5qnrvd6cph5wl7i54lnowb6ltljvbm3fehfrjx2tajq]

Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

Outputs:

resource_id = "ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaa3htpdhd6a5qnrvd6cph5wl7i54lnowb6ltljvbm3fehfrjx2tajq"
resource_name = "steampipetest3479"
tenancy_ocid = "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"

Running SQL query: test-get-query.sql

[
  {
    "display_name": "steampipetest3479",
    "id": "ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaa3htpdhd6a5qnrvd6cph5wl7i54lnowb6ltljvbm3fehfrjx2tajq",
    "lifecycle_state": "ACTIVE"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql

[
  {
    "display_name": "steampipetest3479",
    "id": "ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaa3htpdhd6a5qnrvd6cph5wl7i54lnowb6ltljvbm3fehfrjx2tajq",
    "lifecycle_state": "ACTIVE"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql

null
✔ PASSED

Running SQL query: test-turbot-query.sql

[
  {
    "tenant_id": "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq",
    "title": "steampipetest3479"
  }
]
✔ PASSED

POSTTEST: tests/oci_mysql_db_system

TEARDOWN: tests/oci_mysql_db_system

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  id,
  display_name,
  lifecycle_state as state,
  time_created
from
  oci_mysql_db_system;
+--------------------------------------------------------------------------------------------------+-------------------+---------+---------------------+
| id                                                                                               | display_name      | state   | time_created        |
+--------------------------------------------------------------------------------------------------+-------------------+---------+---------------------+
| ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaa3cgohuiuyfitet2pntq7ni43as4oleew54mo6ii6rlbebigbamhq | steampipetest8206 | DELETED | 2021-07-05 06:40:58 |
| ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaa3htpdhd6a5qnrvd6cph5wl7i54lnowb6ltljvbm3fehfrjx2tajq | steampipetest3479 | DELETED | 2021-07-05 06:55:51 |
| ocid1.mysqldbsystem.oc1.iad.aaaaaaaarsppnnet3yf4xa4l2cdrg277waos4j3vutbyr5wpaintmsze73oq         | test              | ACTIVE  | 2021-07-05 05:19:54 |
+--------------------------------------------------------------------------------------------------+-------------------+---------+---------------------+
> select
  id,
  display_name,
  lifecycle_state as state,
  time_created
from
  oci_mysql_db_system
where
  lifecycle_state <> 'ACTIVE';
+--------------------------------------------------------------------------------------------------+-------------------+---------+---------------------+
| id                                                                                               | display_name      | state   | time_created        |
+--------------------------------------------------------------------------------------------------+-------------------+---------+---------------------+
| ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaa3htpdhd6a5qnrvd6cph5wl7i54lnowb6ltljvbm3fehfrjx2tajq | steampipetest3479 | DELETED | 2021-07-05 06:55:51 |
| ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaa3cgohuiuyfitet2pntq7ni43as4oleew54mo6ii6rlbebigbamhq | steampipetest8206 | DELETED | 2021-07-05 06:40:58 |
+--------------------------------------------------------------------------------------------------+-------------------+---------+---------------------+

> select
  id,
  display_name,
  lifecycle_state as state,
  time_created
from
  oci_mysql_db_system
where
  lifecycle_state = 'ACTIVE'
  and backup_policy -> 'isEnabled' <> 'true';
+----+--------------+-------+--------------+
| id | display_name | state | time_created |
+----+--------------+-------+--------------+
+----+--------------+-------+--------------+
> select
  id,
  display_name,
  lifecycle_state as state,
  time_created
from
  oci_mysql_db_system
where
  lifecycle_state = 'ACTIVE'
  and backup_policy -> 'isEnabled' = 'true';
+--------------------------------------------------------------------------------------------------+-------------------+--------+---------------------+
| id                                                                                               | display_name      | state  | time_created        |
+--------------------------------------------------------------------------------------------------+-------------------+--------+---------------------+
| ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaa5i6il6caoayjfkbfe7eeqzple7khcipxmufenxc6zdkhqcz35azq | test1234          | ACTIVE | 2021-07-06 04:41:50 |
| ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaa7mqkh3fcknfi7glitu3m5fotiyb5qfc7d7dly4kqdhjprjhrl5pa | steampipetest3146 | ACTIVE | 2021-07-06 10:55:16 |
| ocid1.mysqldbsystem.oc1.iad.aaaaaaaarsppnnet3yf4xa4l2cdrg277waos4j3vutbyr5wpaintmsze73oq         | test              | ACTIVE | 2021-07-05 05:19:54 |
+--------------------------------------------------------------------------------------------------+-------------------+--------+---------------------+

```
</details>

### CODE REVIEW

File: docs/tables/oci_mysql_db_system.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+select
+  id,
+  display_name,
+  lifecycle_state as state,
+  time_created
+from
+  oci_mysql_db_system;
+```
+
+### List db systems which are not active
```
Comment (khushboo9024) [inline docs/tables/oci_mysql_db_system.md:19]:
```suggestion
### List DB systems which are not active
```

File: oci/table_oci_mysql_db_system.go [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+	"github.com/oracle/oci-go-sdk/v36/mysql"
+	"github.com/turbot/go-kit/types"
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableMysqlDbSystem(_ context.Context) *plugin.Table {
```
Comment (khushboo9024) [inline oci/table_oci_mysql_db_system.go:17]:
```suggestion
func tableMySQLDBSystem(_ context.Context) *plugin.Table {
```

File: oci/table_oci_mysql_db_system.go [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+	return &plugin.Table{
+		Name:        "oci_mysql_db_system",
+		Description: "OCI Mysql DB System",
```
Comment (khushboo9024) [inline oci/table_oci_mysql_db_system.go:20]:
```suggestion
		Description: "OCI MySQL DB System",
```

File: oci/table_oci_mysql_db_system.go [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+			{
+				Name:        "hostname_label",
+				Description: "The hostname for the primary endpoint of the DB System.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getMysqlDbSytem,
+			},
+			{
+				Name:        "ip_address",
+				Description: "The IP address the DB System is configured to listen on.",
+				Type:        proto.ColumnType_STRING,
```
Comment (khushboo9024) [inline oci/table_oci_mysql_db_system.go:99]: Please check if type can be `ColumnType_IPADDR`.

File: oci/table_oci_mysql_db_system.go [hunk 5]
```diff
@@ -0,0 +1,359 @@
...
+			pagesLeft = false
+		}
+	}
+
+	return nil, err
+}
+
+//// HYDRATE FUNCTION
+
+func getMysqlDbSytem(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```
Comment (khushboo9024) [inline oci/table_oci_mysql_db_system.go:266]:
```suggestion
func getMySqlDBSystem(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```

**Review by khushboo9024 — APPROVED** [review]

**Review by rajlearner17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**